ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartPostHandler.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+package com.github.ambry.frontend.s3;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.github.ambry.commons.ByteBufferReadableStreamChannel;
+import com.github.ambry.commons.Callback;
+import com.github.ambry.frontend.FrontendMetrics;
+import com.github.ambry.frontend.NamedBlobPath;
+import com.github.ambry.frontend.SecurityService;
+import com.github.ambry.rest.RequestPath;
+import com.github.ambry.rest.ResponseStatus;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.rest.RestUtils;
+import com.github.ambry.router.ReadableStreamChannel;
+import java.io.ByteArrayOutputStream;
+import java.nio.ByteBuffer;
+import java.util.GregorianCalendar;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.frontend.FrontendUtils.*;
+import static com.github.ambry.rest.RestUtils.InternalKeys.*;
+
+
+public class S3MultipartPostHandler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(S3MultipartPostHandler.class);
+  private static final String UPLOADS_PARAM_NAME = "uploads";
+  private final SecurityService securityService;
+  private final FrontendMetrics frontendMetrics;
+
+  /**
+   * Construct a handler for handling S3 POST requests during multipart uploads.
+   * @param securityService   the {@link SecurityService} to use.
+   * @param frontendMetrics   {@link FrontendMetrics} instance where metrics should be recorded.
+   */
+  public S3MultipartPostHandler(SecurityService securityService, FrontendMetrics frontendMetrics) {
```

Comment:
Perhaps sufficient to call it `S3MultipartUploadHandler`? the HTTP method POST isn't significant here. 

Alternatively - would it make sense to put all multipart stuff in one class, and separate them in individual inner classes, like Create-, Abort-, CompleteMultipartUpload, and UploadPart? Also, I suggest to name class according naming to S3 spec, it makes it easier to understand later.